test2_2/list.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+struct SortedListElement;
+
+struct SortedList;

```

Comment:
```suggestion
typedef struct SortedList SortedList;
```
При таком описании абстрактного типа нет необходимости писать `struct` при каждом упоминании типа. Достаточно указывать только имя.